nexosim/src/ports/sink/blocking_event_queue.rs

```diff
@@ -0,0 +1,143 @@
+use std::fmt;
+use std::iter::FusedIterator;
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::mpsc::{channel, Receiver, Sender};
+use std::sync::Arc;
+
+use super::{EventSink, EventSinkStream, EventSinkWriter};
+
+/// A blocking event queue with an unbounded size.
+///
+/// Implements [`EventSink`], while [`EventSinkStream`] is implemented by
+/// [`BlockingEventQueueReader`] available through the
+/// [`BlockingEventQueue::reader`] method.
+pub struct BlockingEventQueue<T> {
+    is_open: Arc<AtomicBool>,
+    sender: Sender<T>,
+    receiver: Receiver<T>,
+}
+
+impl<T> BlockingEventQueue<T> {
+    /// Creates an open `BlockingEventQueue`.
+    pub fn new() -> Self {
+        Self::new_with_open_state(true)
+    }
+
+    /// Creates a closed `BlockingEventQueue`.
+    pub fn new_closed() -> Self {
+        Self::new_with_open_state(false)
+    }
+
+    /// Returns consumer handle.
+    pub fn reader(self) -> BlockingEventQueueReader<T> {
```

Comment:
I would rather call it `into_reader` to make it clear it is consuming `self`, especially that the `writer` method does not.